### ISSUE TITLE
Fix autofix erroneous changes for float property

### ISSUE THREAD

**Issue by wilhen01** [stylelint/stylelint#4490]
> Clearly describe the bug

I'm using Stylelint with a typescript project that contains styled components using the `@emotion/styled`  library. When running `styelint --fix` on the codebase, instances of `float` are incorrectly changed to `cssFloat`, which breaks the styling. Weirdly, no warnings are emitted, either with or without the fix flag. 

> Which rule, if any, is the bug related to?

Not sure what rule causes this behaviour, haven't had any luck tracking it down in the docs.

> What CSS is needed to reproduce the bug?

Example styled component

```ts
const StyledCloseBtn = styled(Btn)<Btn.propTypes>({
  float: 'right',
  minWidth: 'unset !important',
  position: 'relative',
  top: -2,
  svg: { fill: 'currentColor', width: 16, height: 16 }
});
```

after running the fix command it changes to:

```ts
const StyledCloseBtn = styled(Btn)<Btn.propTypes>({
  cssFloat: 'right',
  minWidth: 'unset !important',
  position: 'relative',
  top: -2,
  svg: { fill: 'currentColor', width: 16, height: 16 }
});
```

> What stylelint configuration is needed to reproduce the bug?

Complete config below

```json
module.exports = {
  extends: [
    'stylelint-config-standard',
    'stylelint-a11y/recommended',
    'stylelint-config-prettier'
  ],
  plugins: ['stylelint-no-unsupported-browser-features'],
  rules: {
    'plugin/no-unsupported-browser-features': [
      true,
      {
        browsers: ['last 1 edge version']
      }
    ],
    'color-named': 'never',
    'color-hex-case': 'upper',
    'color-hex-length': 'long',
    'declaration-no-important': true,
    'function-parentheses-space-inside': 'never',
    'function-comma-space-before': 'never',
    'function-comma-space-after': 'always',
    'declaration-empty-line-before': 'never',
    'declaration-block-single-line-max-declarations': null
  }
};

```

> Which version of stylelint are you using?

`12.0.0` (also occurs on 10.1.0, upgrade made no difference)

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint ./src --fix`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, styled components

> What did you expect to happen?

No changes to be made to the file

> What actually happened (e.g. what warnings or errors did you get)?

float changed to cssFloat. No warnings are emitted, either with or without the fix flag.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Most likely problem in `postcss-jsx` parser, which we use for CSS-in-JS.

**Comment by jeddy3** [comment]
Labeling as "help wanted" if anyone has time to fix the issue upstream in the `postcss-jsx` parser.

**Comment by jeddy3** [comment]
We've forked the parser we use for CSS-in-JS [into stylelint organisation](https://github.com/stylelint/postcss-css-in-js), and we're looking for help to fix our fork to close this issue.

See https://github.com/stylelint/stylelint/issues/4574.

**Comment by jeddy3** [comment]
I've labelled this as "upstream" as this is an issue [in one of the syntaxes stylelint uses](https://stylelint.io/about/syntaxes). Please consider [contributing to that syntax](https://stylelint.io/developer-guide/syntaxes) if you'd like the see this issue resolved fully.

If you're unable to contribute to the syntax, please consider [contributing](https://stylelint.io/contributing) a [workaround to stylelint itself](https://stylelint.io/developer-guide/syntaxes#add-a-workaround) so that it won't attempt to autofix sources that contain `float` properties in JavaScript objects.

**Comment by guoyunhe** [comment]
I am having the same issue. Have to disable stylelint to make my code work...

**Comment by alzalabany** [comment]
this is still active problem :/;
to avoid i had to do

```
const styles = { };
styles.float = 'right';

<a style={ styles }> ....</a>
```

just to note, this is problem not just with styled-components; i'm facing this in jsx in general

**Comment by Avivhdr** [comment]
Any progress with @rbong's PR?

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.

**Comment by hudochenkov** [comment]
Fix released in `@stylelint/postcss-css-in-js@0.38.0`.
